smash/core/signal_analysis/prcp_indices/prcp_indices.py

```diff
@@ -14,56 +14,106 @@
 __all__ = ["PrecipitationIndices", "precipitation_indices"]
 
 
-class PrecipitationIndices(dict):
+class PrecipitationIndices:
+
     """
     Represents precipitation indices computation result.
 
-    Notes
-    -----
-    This class is essentially a subclass of dict with attribute accessors.
-
-    TODO FC: Fill
+    Attributes
+    ----------
+    std : numpy.ndarray
+        The precipitation spatial standard deviation.
+    d1 : numpy.ndarray
+        The first scaled moment :cite:p:`zocatelli_2011`.
+    d2 : numpy.ndarray
+        The second scaled moment :cite:p:`zocatelli_2011`.
+    vg : numpy.ndarray
+        The vertical gap :cite:p:`emmanuel_2015`.
+    hg : numpy.ndarray
+        The horizontal gap (TODO FC: Fill citation).
 
+    See Also
+    --------
+    precipitation_indices : Compute precipitation indices of the Model object.
     """
 
-    def __getattr__(self, name):
-        try:
-            return self[name]
-        except KeyError as e:
-            raise AttributeError(name) from e
+    def __init__(self, data: dict | None = None):
+        if data is None:
+            data = {}
+
+        self.__dict__.update(data)
+
+    def to_numpy(self, axis=0):
+        """
+        Convert the `PrecipitationIndices` object to a numpy.ndarray.
+
+        The attribute arrays are stacked along a user-specified axis of the resulting array in alphabetical order
+        based on the names of the precipitation indices.
+
+        Parameters
+        ----------
+        axis : int, default 0
+            The axis along which the precipitation arrays will be joined.
+
+        Returns
+        -------
+        res : numpy.ndarray
+            The `PrecipitationIndices` object as a numpy.ndarray.
 
-    __setattr__ = dict.__setitem__
-    __delattr__ = dict.__delitem__
+        Examples
+        --------
+        >>> import smash
+        >>> from smash.factory import load_dataset
+        >>> setup, mesh = load_dataset("cance")
+        >>> model = smash.Model(setup, mesh)
+        >>> prcpind = smash.precipitation_indices(model)
 
-    def __repr__(self):
-        if self.keys():
-            m = max(map(len, list(self.keys()))) + 1
-            return "\n".join(
-                [
-                    k.rjust(m) + ": " + repr(v)
-                    for k, v in sorted(self.items())
-                    if not k.startswith("_")
-                ]
-            )
-        else:
-            return self.__class__.__name__ + "()"
```

Comment:
Je garderais une representation des objets parce que la tu fais print(prcpind) tu as plus d'info. On peut juste faire un truc comme ca 
`def  \_\_repr\_\_(self):
        
        dct = self.__dict__
        
        if dct.keys():
            m = max(map(len, list(dct.keys()))) + 1
            return "\n".join(
                [
                    k.rjust(m) + ": " + repr(v)
                    for k, v in sorted(dct.items())
                    if not k.startswith("_")
                ]
            )
        else:
            return self.__class__.__name__ + "()"`

Reply:
Oui oui, du coup j'ai mis un repr(type(v)) au lieu de la représentation de tous les valeurs.